lib/dataContract/DataContractFactory.js

```diff
@@ -59,9 +61,17 @@ class DataContractFactory {
    * @return {DataContract}
    */
   createFromSerialized(payload, options = { }) {
-    const rawDataContract = decode(payload);
+    try {
+      const rawDataContract = decode(payload);
 
-    return this.createFromObject(rawDataContract, options);
+      return this.createFromObject(rawDataContract, options);
+    } catch (error) {
+      throw new SerializedObjectParsingError(
```

Comment:
I would wrap this error to `InvalidDataContractError` as we are doing if Data Contract is not valid https://github.com/dashevo/js-dpp/blob/222c625c23d2df36db7815fa36438dbfd6047a3e/lib/dataContract/DataContractFactory.js#L43